python/ray/experimental/channel/shared_memory_channel.py

```diff
@@ -750,6 +777,10 @@ def read(self, timeout: Optional[float] = None) -> Any:
         actor_id = self._get_self_actor_id()
         return self._channel_dict[actor_id].read(timeout)
 
+    def release_buffer(self, timeout: Optional[float] = None):
+        self.ensure_registered_as_reader()
+        self._channel_dict[self._get_self_actor_id()].release_buffer(timeout)
```

Comment:
Is it possible that the channel is an IntraProcessChannel? I think it's unlikely, but I just want to double-check.

Reply:
Afaik, communications between actors are conducted through `IntraProcessChannel`, and communications between the driver and actor go through `BufferedSharedMemoryChannel`. Destruction of `CompiledDAGRef` happens on the driver, and the deserialization I skipped occurs between the driver and an actor. Therefore, the underlying channel is always `BufferedSharedMemoryChannel`.

I stepped into some unit tests where `IntraProcessChannel` is used (e.g. `TestCompositeChannel::test_composite_channel_one_actor`) and verified that DAG only uses `BufferedSharedMemoryChannel` to write its outputs.

Let me know if I misunderstood anything! I'm still in the process of getting more familiar with the Ray backend :)

Reply:
It is correct that the channels between the driver process and actors must be shared memory channels.